### PR TITLE
merge release scripts, tag each shortbreaks release as n1-latest

### PR DESCRIPTION
**Pull request by hxpaul**
#### What are the links to relevant tickets?

https://hxshortbreaks.atlassian.net/browse/WEB-7736
#### What does this PR do?

merges createPrivateRelease.sh with short breaks' postRelease.sh so that when we run postRelease.sh it will tag the latest release for this version as vn-latest so we can install the latest release for any major version. This means we can safely require the latest v1 of a repo without worry about breaking changes.

This brings short breaks and core into alignment in this area too 💒 
#### What functional/unit tests does this PR have?

none
#### Is there anything a developer should be aware of when reviewing this?

you can run this locally with either `npm run release production 1` or `./nodeApps/postRelease.sh production 1` - ideally it would be run always as an npm script it's cleaner then, but leaving the fall back in for now.

To run this you need GHUSER or GITHUB_USER and GHPASS or GITHUB_API_TOKEN env vars (optionally plus others for short breaks graphite logging). Also you'd need to override the release branch so and you're debugging so don't actually `curl` anything just echo instead

```
DEBUG=true RELEASE_BRANCH=lunchtime-larks GITHUB_USER=foo GITHUB_API_TOKEN=bar npm run release
```

where `lunchtime-larks` is this branch.

It's a simple change but it would be nice to be able to do npm style `^` requiring of our github repos in short breaks and it brings sb + core closer together (group hug).

To simplify this and move to a single set of variables we need to either move from `GHUSER` and `GHPASS` to `GITHUB_USER` and `GITHUB_API_TOKEN` in each repo that uses the original postRelease.sh - this would carry on working fine without that migration though.

---
#### Quality checklist (for PR author to complete BEFORE code review):
- [ ] I've checked there is appropriate unit test coverage.
- [ ] I've updated documentation with any changes to procedures.
- [x] I've checked this work against the requirements of the Jira.
- [ ] This change passes all necessary tests (cross browser, automated or otherwise).
- [x] I am ready for this to be code reviewed, merged and tested.
#### Reviewer 1
- [x] I agree with the assumptions made in the quality checklist.
- [ ] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!
#### Reviewer 2
- [x] I agree with the assumptions made in the quality checklist.
- [ ] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

### CODE REVIEW

**Comment by hx-markterry** [comment]
I'd rather add updates to the existing script to gather args when running from npm maybe via a command line switch `--npm` or similar.

**Comment by hx-markterry** [comment]
We need to move away from specifying the github password as it prevents against having 2fa on an account.

**Comment by hxpaul** [comment]
regarding the points above, I've implemented the changes this way so that the short breaks ci will carry on working, but using the core script.

**Comment by hx-markterry** [comment]
ok from me

**Comment by robhuzzey** [comment]
@hxpaul what is happening with this, I would find this very useful :)

File: nodeApps/createPrivateRelease.sh [hunk 1]
```diff
@@ -4,13 +4,28 @@
...
 set -e
 
+ENVIRONMENT=$1
+NO_METRIC=$2
+if [ $DEBUG != "" ]; then
+  SAFE_DEBUG_ECHO=echo
+fi
+
+# allow the shortbreaks' version $GHUSER:$GHPASS so ci keeps running
+if [ -z $GITHUB_USER ]; then
```
Comment (robhuzzey) [inline nodeApps/createPrivateRelease.sh]: I would like to see us using common Variables across the group for things like this. Perhaps something for the culture repo? For now, this is good for backwards compatibility.

**Comment by robhuzzey** [comment]
I've given a +1 to this too. @hx-markterry any chance we can see this "resurrected" as it addresses some issues I have been facing using CI in SB?

**Comment by notABluesSinger** [comment]
I am 100% for us merging and using the same process here. It makes perfect sense as long as our requirements don't significantly differ.
I do however have some questions before we can start using in SB;

1. @hx-markterry, what was the reason for forking from our `postInstall` file rather than simply expanding on the original functionality. I want to understand what was different enough to prevent using the same script, so that I can be sure that it fits with our roadmap and what is published in the culture repo.
2. @robhuzzey, can you please detail the issues that this addresses. It'd be good to have on this PR.
3. Can we all ensure that this is treated as group tech and, as such, no changes are merged without approval from all sides of the business that are using it.
4. @hx-markterry, I'd also like to hear your thoughts on the discussion about moving over to node scripts for this kind of processing. Is that likely to come in and effect this particular functionality anytime soon?

I'd also like to see a detailed breakdown of what using this joined script would do / produce, including optional paths.

Also, could you document the changes required to move to a single set of variables, as I think the mix and match causes some confusion.

**Comment by hxpaul** [comment]
@notABluesSinger I've updated the README in this repo with what it does and the optional variables

**Comment by robhuzzey** [comment]
@notABluesSinger sure... the 2 issues it addressed for me was:

1) Ability to release on a branch other than master
2) Github user & pass variable names were different for distribution to SB.

**Comment by hx-markterry** [comment]
The main reason for moving to user and token rather than user and pass is to support accounts with 2FA.

**Comment by hx-markterry** [comment]
This was a separate script initially as it eased the process of using a single repo for these scripts, it seemed different enough at the time to maintain 2 smaller scripts for different CIs, but the work here has been done to merge them, which will be easier to maintain in my opinion.

File: nodeApps/postRelease.sh [hunk 2]
```diff
@@ -1,15 +1,7 @@
...
-if [ $ENVIRONMENT = "production" ] || [ $ENVIRONMENT = "master" ]; then
-	# Tag this deploy and make a release in github
-	curl --user "${GHUSER}:${GHPASS}" --data "{\"tag_name\": \"v${CANDIDATE_VERSION}\",\"target_commitish\": \"master\",\"name\": \"v${CANDIDATE_VERSION}\",\"body\": \"Production release by @${CIRCLE_USERNAME} (via build: [${CIRCLE_BUILD_NUM}](https://circleci.com/gh/${REPO_OWNER}/${CIRCLE_PROJECT_REPONAME}/${CIRCLE_BUILD_NUM})).\",\"draft\": false,\"prerelease\": false}" https://api.github.com/repos/${REPO_OWNER}/${CIRCLE_PROJECT_REPONAME}/releases
-fi
-# Add a deployment counter in the metrics platform (graphite)
-if [ -z $NO_METRIC ]; then
-	echo "${GRAPHITE_API_KEY}.counters.${APP_NAME}.${ENVIRONMENT}.inf.deployment 1" | nc ${GRAPHITE_ENDPOINT_PREFIX}.carbon.hostedgraphite.com 2003
-fi
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+echo $0 is depracated please update to createPrivateRelease.sh instead
```
Comment (kevinhodges) [inline nodeApps/postRelease.sh]: typo on `deprecated`

File: README.md [hunk 3]
```diff
@@ -6,7 +6,38 @@ This project holds various deployment scripts that are used with AWS CodeDeploy
 
 ### `createPrivateRelease.sh`
 
-This will check if the `package.json` version has been updated for a project and then create a new release on github after running `npm build` and commiting any changed release assets in the `dist` directory.
+This will check if the `package.json` version has been updated for a project and then create a new release on github after running `npm build` and commiting any changed release assets in the `dist` directory.A
```
Comment (kevinhodges) [inline README.md]: you've got a random `A` at the end of this line

File: nodeApps/createPrivateRelease.sh [hunk 4]
```diff
@@ -88,17 +112,25 @@ echo "Latest commit is: ${CURRENT_SHA}"
...
 # Create a new updated tag
-curl -u "${GITHUB_USER}:${GITHUB_API_TOKEN}" --data "{\"ref\": \"refs/tags/v${THIS_MAJOR_VERSION}-latest\",\"sha\": \"${CURRENT_SHA}\"}" https://api.github.com/repos/${APP_NAME}/git/refs
+$SAFE_DEBUG_ECHO curl -u "${GITHUB_USER}:${GITHUB_API_TOKEN}" --data "{\"ref\": \"refs/tags/v${THIS_MAJOR_VERSION}-latest\",\"sha\": \"${CURRENT_SHA}\"}" https://api.github.com/repos/${APP_NAME}/git/refs
+
+# Add a deployment counter in the metrics platform (graphite)
+if [ "$ENVIRONMENT" != "" } && [ "${GRAPHITE_API_KEY}" != "" ] && [ -z $NO_METRIC ]; then
+  if [ $DEBUG != "" ]; then
+	  echo "${GRAPHITE_API_KEY}.counters.${BASE_NAME}.${ENVIRONMENT}.inf.deployment 1 |"
+  fi
+	echo "${GRAPHITE_API_KEY}.counters.${BASE_NAME}.${ENVIRONMENT}.inf.deployment 1" | $SAFE_DEBUG_ECHO nc ${GRAPHITE_ENDPOINT_PREFIX}.carbon.hostedgraphite.com 2003
```
Comment (kevinhodges) [inline nodeApps/createPrivateRelease.sh]: indentation not quite right here

File: nodeApps/createPrivateRelease.sh [hunk 5]
```diff
@@ -88,17 +112,25 @@ echo "Latest commit is: ${CURRENT_SHA}"
 # Tag this deploy and make a release in github
 RELEASE_JSON="{\"tag_name\": \"v${THIS_VERSION}\",\"target_commitish\": \"${CURRENT_SHA}\",\"name\": \"v${THIS_VERSION}\",\"body\": \"${MESSAGE}\",\"draft\": false,\"prerelease\": false}"
```
Comment (kevinhodges) [inline nodeApps/createPrivateRelease.sh]: `$(MESSAGE}` doesn't include `${CIRCLE_USERNAME}` which is really useful for seeing who deployed a given release, can we add that in please.

e.g. https://github.com/holidayextras/revolver/releases/tag/v6.29.0

Comment (notABluesSinger) [inline nodeApps/createPrivateRelease.sh]: Was also nice to have the link back to the build from the release... nice easy way to create an audit trail

Comment (hx-markterry) [inline nodeApps/createPrivateRelease.sh]: The link back is there: https://github.com/holidayextras/deployment-helpers/pull/10/files#diff-efc43368f159a2d0209883f8af2537a9R103

Comment (notABluesSinger) [inline nodeApps/createPrivateRelease.sh]: oh yeah, so it is :)

**Comment by hpoom** [comment]
@hxpaul mentioned this was blocked on Slack. Is this still blocked and if so what by?

**Comment by hxpaul** [comment]
I didn't like that there were two ways of doing this, so I made a third way, feel free to contribute to release.js

too many standards

Merry christmas!

**Pull request closed without merging**